components/dashboards/org.wso2.carbon.dashboards.api/src/main/java/org/wso2/carbon/dashboards/api/internal/DashboardRestApi.java

```diff
@@ -182,4 +186,66 @@ public Response delete(@PathParam("id") String id) {
             return Response.serverError().entity("Cannot delete dashboard '" + id + "'.").build();
         }
     }
+
+    @GET
+    @Path("/roles")
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getRoles() {
+        try {
+            List<Role> allRoles = dashboardDataProvider.getAllRoles();
+            return Response.ok()
+                    .entity(allRoles)
+                    .type(MediaType.APPLICATION_JSON)
+                    .build();
+        } catch (DashboardException e) {
+            LOGGER.error("An error occurred while retrieving roles.", e);
+            return Response.serverError()
+                    .entity("An error occurred while retrieving roles.")
```

Comment:
Instead of this error message, shall we use `Cannot retrieve user roles.`. Its more user-friendly for the APi consumer.
Please correct here & similar places.